### PR TITLE
Feat: LSP will execute 'kpm metadata' to get the localpath of the kcl dependecies.

### LINKED ISSUE #543: Enhancement: `LSP` needs to request `kpm` to get the path of external dependencies.
## `LSP` needs to request `kpm` to get the path of external dependencies.
`kpm` will support command `kpm metadata` which returns the metadata includes the external dependency local path of the current kcl package. `LSP` can get the external package path from the metadata. 

[More information about `kpm metadata`](https://github.com/KusionStack/kpm/issues/66)

In the process of compiling, `LSP` will call `lookup_compile_unit()` to find the parts that need to be compiled. `lookup_compile_unit()` will return the files that need to be compiled and the compilation options.

Therefore, the process of executing `kpm metadata` to obtain the external package paths in method `lookup_compile_unit()`, and add the result to the compilation options to return.

```rust
// The pseudo code
fn lookup_compile_unit(
    file: &str,
    load_pkg: bool,
) -> (Vec<String>, Option<LoadProgramOptions>) {

        let opts := LoadProgramOptions::new();

        // 1. get 'kcl.mod' file path
        kcl_mod_path := lookup_kcl_mod_file(file)

        // 2. execute 'kpm metadata' under 'kcl.mod' file path
        metadata := Command.run("kpm", "metadata").under_dir(kcl_mod_path)

        // 3. fill the dependencies map in 'LoadProgramOptions'.
        fill_dependencies_map(metadata, opts)

        ......
}
```

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix #543 

#### 2. What is the scope of this PR (e.g. component or file name):

LSP

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [x] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

LSP will execute 'kpm metadata' to get the localpath of the kcl dependecies.

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

#### 6. Release note

Please refer to [Release Notes Language Style Guide](https://kusionstack.io/docs/governance/release-policy/) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: kclvm/ast/src/ast.rs [hunk 1]
```diff
@@ -428,6 +428,7 @@ pub struct ImportStmt {
     pub rawpath: String,
     pub name: String,
     pub asname: Option<String>,
+    pub pkg_name: String,
```
Comment (Peefy) [inline kclvm/ast/src/ast.rs:431]: What does this attribute mean?

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 2]
```diff
@@ -126,8 +127,38 @@ fn goto_def_for_import(
     program: &Program,
 ) -> Option<GotoDefinitionResponse> {
     let pkgpath = &stmt.path;
-    let real_path =
+    let mut real_path =
         Path::new(&program.root).join(pkgpath.replace('.', &std::path::MAIN_SEPARATOR.to_string()));
+    let mut positions = get_pos_from_real_path(&real_path);
+
+    if positions.len() == 0 && !real_path.exists() {
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:134]: `positions.is_empty()`. Besides, run `cargo fix`

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 3]
```diff
@@ -126,8 +127,38 @@ fn goto_def_for_import(
...
+        real_path = PathBuf::new();
+
+        let pkg_root = fetch_metadata(program.root.clone().into())
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:137]: Extract the logic for processing metadata into a smaller function for the convenience of other LSP functions.

File: kclvm/tools/src/LSP/src/goto_def.rs [hunk 4]
```diff
@@ -126,8 +127,38 @@ fn goto_def_for_import(
...
+            Err(_) => String::new(),
+        };
+        pkgpath.split('.').for_each(|s| real_path.push(s));
+    }
+    positions = get_pos_from_real_path(&real_path);
+
+    positions_to_goto_def_resp(&positions)
+}
+
+fn get_pos_from_real_path(
```
Comment (Peefy) [inline kclvm/tools/src/LSP/src/goto_def.rs:159]: Move this function into the util crates.

File: kclvm/tools/src/LSP/src/goto_def.rs (hunk 3 above)
Comment (zong-zhe) [reply kclvm/tools/src/LSP/src/goto_def.rs]: fixed.

File: kclvm/tools/src/LSP/src/goto_def.rs (hunk 4 above)
Comment (zong-zhe) [reply kclvm/tools/src/LSP/src/goto_def.rs]: fixed.

File: kclvm/tools/src/LSP/src/goto_def.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/tools/src/LSP/src/goto_def.rs]: fixed.

File: kclvm/ast/src/ast.rs (hunk 1 above)
Comment (zong-zhe) [reply kclvm/ast/src/ast.rs:434]: `pkg_name` means the name of the package that the current import statement indexs to. If the current import statement indexs to the current package, `pkg_name`  is "\__main\__",  If the current import statement indexs to the external packages, `pkg_name` is the name of the package.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**